### PR TITLE
ci: Change Unity UnitTest from Mono to IL2CPP and enabled on CI

### PR DESCRIPTION
**Pull request by guitarrapc**
## tl;dr;

Previously Unity's Unit Test was commented out due to Unity Healess (DedicatedServer) API change was unstable. I've confirm it's stable now and re-enable on GitHub Actions.

To Run UnitTest IL2CPP.

* Bump Unity version from 2020.2.1f1 to 2022.3.39f1. (LTS)
* Change Unity versions running on matrix to latest LTS.
* Re-enable Unity UnitTest on GitHub Actions
* Change Unity UnitTest from Mono to IL2CPP
* Change RuntimeUnitTestToolkit from UnityPackage to UPM
* Bump RuntimeUnitTestToolkit from 2.3.0 to 2.6.0

For CI re-run usability improvement.
* Split Unity UnitTest from build, as IL2CPP build takes over 13min (3min for Mono)

### CODE REVIEW

**Comment by timcassell** [comment]
FYI, unity-test-runner supports standalone test mode now for a while. If you want to, you could just use that instead of doing separate build + test steps. https://game.ci/docs/github/test-runner#testmode

**Comment by guitarrapc** [comment]
@timcassell Thank you for interesting information. Currently Cysharp is using it's own package RuntimeUnitTestToolkit and run it as binary for Dog-fooding. I've added test-runner to watchlist.
